src/components/FeatureViews/AdminPortal/BusinessDetails.js

```diff
@@ -0,0 +1,452 @@
+    query {
+        business {
+            id
+            name
+            phoneNumber
+            email
+            address
+            availabilityList {
+                dayOfWeek
+                startTime
+                endTime
+            }
+        }
+    }
+`;
+
+const UPDATE_BUSINESS = gql`
+    mutation MyMutation(
+        $address: String
+        $email: String
+        $name: String
+        $phoneNumber: String
+    ) {
+        updateBusiness(
+            address: $address
+            email: $email
+            name: $name
+            phoneNumber: $phoneNumber
+        ) {
+            business {
+                id
+                name
+                address
+                email
+                phoneNumber
+            }
+            updated
+        }
+    }
+`;
+
+const BusinessDetails = () => {
+    const classes = useStyles();
+    const [activeStep, setActiveStep] = useState(0);
+    const [updateData, setUpdateData] = useState({
+        businessName: '',
+        businessPhone: '',
+        businessEmail: '',
+        businessAddress: '',
+    });
+
+    const [submitData] = useMutation(UPDATE_BUSINESS, {
+        onCompleted: () => {
+            handleBackStep();
+        },
+        update: (cache, data) => {
+            const updatedBusiness = data.data.updateBusiness.business;
+            cache.writeQuery({
+                query: GET_BUSINESS,
+                data: {
+                    business: updatedBusiness,
+                },
+            });
+        },
+    });
+
+    const handleStepChange = () => setActiveStep((prevState) => prevState + 1);
+    const handleBackStep = () => setActiveStep((prevState) => prevState - 1);
+
+    const handleOnChange = (e) => {
+        const { name, value } = e.target;
+        setUpdateData((prevState) => ({
+            ...prevState,
+            [name]: value,
+        }));
+    };
+
+    const onSubmit = () => {
+        submitData({
+            variables: {
+                name: updateData.businessName,
+                phone: updateData.businessPhone,
+                email: updateData.email,
+                address: updateData.businessAddress,
+            },
+        });
+    };
+
+    const { loading, error, data } = useQuery(GET_BUSINESS, {
+        onCompleted: () => {
+            setUpdateData({
+                businessName: business.name,
+                businessPhone: business.phoneNumber,
+                businessEmail: business.email,
+                businessAddress: business.address,
+            });
+        },
+    });
+
+    if (loading) {
+        return <Loading />;
+    }
+    if (error) {
+        return (
+            <Typography>
+                There has been an error! Error: {error.message}
+            </Typography>
+        );
+    }
+
+    const { business } = data;
+
+    const getBusinessDetailsContent = (step) => {
+        switch (step) {
+            case 0:
+                return (
+                    <Grid
```

Comment:
@pswk1 separate these out into 2 different components (one ViewBusinessDetails and EditBusinessDetails)